backend/src/main/java/com/bakdata/conquery/models/messages/network/NetworkMessageContext.java

```diff
@@ -12,13 +12,15 @@
 import com.bakdata.conquery.models.worker.Workers;
 import lombok.Getter;
 
+@Getter
 public abstract class NetworkMessageContext<MESSAGE extends NetworkMessage<?>> extends MessageSender.Simple<MESSAGE> {
-	@Getter
 	private final JobManager jobManager;
+	private final int backpressure;
```

Comment:
Der backpressure ist jetzt nur vom Manager benutzt aber auch auf dem Shard verfügbar. Sollte der dort nicht auch geprüft werden auch wenn dort nie so viele Dateien übrig sind.

Reply:
Der backpressure wird sogar nur freiwillig eingehalten via waitForFreeJobQueue, die Info über die größe der JobQueue haben die Shards ja auch gar nicht. Dafür gibt es ja schon den Backpressure in der Queue, der für große Nachrichten ist. Das ganze ist wie gesagt auch echt nur eine zwischenlösung damit wir Zugriff auf die Info haben. Bei einem HTTP Backend wäre das alles deutlich unkomplizierter.